File: demos/ios/Demos/Demos/views/BaseLineChart.m

#import "BaseLineChart.h"

@interface BaseLineChart ()

@property (nonatomic, strong) F2CanvasView *canvasView;
@property (nonatomic, strong) F2Chart *chart;
@end

@implementation BaseLineChart {
    CGRect _frame;
}

- (instancetype)init {
    return [self initWithFrame:CGRectMake(0, 0, [[UIScreen mainScreen] bounds].size.width, 280)];
}

- (instancetype)initWithFrame:(CGRect)frame {
    if(self = [super initWithFrame:frame]) {
        F2CanvasView *view = [[F2CanvasView alloc] initWithFrame:frame];
        view.backgroundColor = [UIColor clearColor];
        self.canvasView = view;
        [self chartRender];
        [self addSubview:self.canvasView];
    }
    return self;
}

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_baseLine" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    NSArray *jsonArray = [F2Utils toJsonArray:jsonData];
    self.chart.clear();
    self.chart.canvas(self.canvasView).padding(10, 10, 10, 10.f).source(jsonArray);
    self.chart.scale(@"date", @{@"tickCount": @(3)});
    self.chart.scale(@"value", @{@"nice": @(YES)});
    self.chart.axis(@"date", @{
        @"grid": @(NO),
        @"label": @{           
            @"item": [F2Callback callback:^NSDictionary *_Nonnull(NSDictionary *_Nonnull param) {
                NSInteger index = [[param objectForKey:@"index"] integerValue];
                NSArray *textAlign = @[@"start", @"center", @"end"];                
                return @{@"textColor": index % 2 == 0 ? @"#000000" : @"#DC143C", @"textAlign": textAlign[index]};
            }]
        }
    });
    self.chart.axis(@"value", @{@"grid": @{@"type": @"dash", @"dash": @[@(15), @(15), @(5), @(5)]}});
    self.chart.line().position(@"date*value").fixedSize(2);

    self.chart.tooltip(@{});
    
    self.chart.guide().background(@{@"color":@"#FF0000", @"opacity":@0.2, @"leftBottom":@[@"min", @"min"], @"rightTop":@[@"max", @(80)]});
    self.chart.guide().background(@{@"color":@"#00FF00",@"opacity":@0.2, @"leftBottom":@[@"min", @(80)], @"rightTop":@[@"max", @(320)]});
    
    self.chart.animate(YES);
    
    self.chart.render();
}

- (F2Chart *)chart {
    if(!_chart) {
        _chart = [F2Chart chart:self.canvasView.frame.size name:NSStringFromClass([self class])];
    }
    return _chart;
}

- (NSString *)name {
    return NSStringFromClass([self class]);
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/BaseLineChart.swift
//
//  BaseLineUIView.swift
//  Demos-Swift
//
//  Created by weiqing.twq on 2022/6/1.
//

import UIKit
import F2

class BaseLineChart: UIView {
    
    public var canvasView:F2CanvasView?
    public var chart:F2Chart?
    
    required override init(frame: CGRect) {
        super.init(frame: frame)
        self.canvasView = F2CanvasView.canvas(frame)
        self.addSubview(self.canvasView!)
        self.chart = F2Chart.init(self.canvasView!.bounds.size, name: "swiftdemo")
        self.chartRender();
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func chartRender() {
        let jsonPath = Bundle.main.path(forResource: "Res/mockData_baseLine", ofType: "json")
        guard let jsonString = try? String.init(contentsOfFile: jsonPath!) else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonString)
        self.chart!.canvas()(self.canvasView!)
        self.chart?.padding()(10, 20, 10, 0)
        self.chart!.source()(jsonData)
        self.chart!.line()().position()("date*value")
        self.chart!.scale()("date", ["tickCount":3])
        let itemBlock = {(param: Dictionary<AnyHashable, Any>) -> Dictionary<AnyHashable, Any> in
            let index = param["index"] as? Int
            if (index! % 2 != 0) {
                return ["textColor": "#DC143C"]
            }
            return param
        } as ItemCallback
        let itemCallback = F2Callback.init(itemBlock);
        self.chart!.axis()("date",
                           ["grid": false,
                           "label":["item": itemCallback]]);
        self.chart!.animate()(true)
        self.chart!.render()();
    }
}
